### PR TITLE
fix: enable authorization cache by default

### PR DESCRIPTION
**Pull request by id**
Fixes https://emqx.atlassian.net/browse/EMQX-8459.

Pls let me know if this is a wrong way to approach this issue.

### CODE REVIEW

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
Maybe set default value in schema 

add changelog

**Comment by id** [comment]
@zmstone DeepL-ed Chinese version, can't vouch for quality.

In the schema it should already be enabled by default? https://github.com/emqx/emqx/blob/master/apps/emqx/src/emqx_schema.erl#L340

File: changes/v5.0.14-zh.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# v5.0.14
+
+## 增强
+
+- 默认启用授权缓存 [#9626](https://github.com/emqx/emqx/pull/9626).
```
Comment (lafirest) [inline changes/v5.0.14-zh.md:5]: - 默认启用授权缓存 [#9626](https://github.com/emqx/emqx/pull/9626)。

File: changes/v5.0.14-en.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# v5.0.14
+
+## Enhancements
+
+- Enable authorization cache by default [#9626](https://github.com/emqx/emqx/pull/9626).
```
Comment (zmstone) [inline changes/v5.0.14-en.md:5]:
```suggestion
- Return authorization settings with default values [#9626](https://github.com/emqx/emqx/pull/9626).
  The cache is enabled by default, but due to the missing default value in `GET` response of `/authorization/settings`, it seemed to be disabled from the dashboard.
```

File: changes/v5.0.14-zh.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# v5.0.14
+
+## 增强
+
+- 默认启用授权缓存 [#9626](https://github.com/emqx/emqx/pull/9626)。
```
Comment (zmstone) [inline changes/v5.0.14-zh.md:5]:
```suggestion
- 为授权设置 API 返回默认值 [#9626](https://github.com/emqx/emqx/pull/9626)。
  授权缓存默认为开启，但是在此修复前，因为默认值在 `/authorization/settings` 这个 API 的返回值中缺失，
  使得在仪表盘配置页面中看起来是关闭了。
```

**Review by zmstone — COMMENTED** [review]
The fix looks good (I mean it's good to have the default value explicitly set in config file).
However, the proper fix is this
```
diff --git a/apps/emqx_authz/src/emqx_authz_api_settings.erl b/apps/emqx_authz/src/emqx_authz_api_settings.erl
index 905dbe223..c991bde83 100644
--- a/apps/emqx_authz/src/emqx_authz_api_settings.erl
+++ b/apps/emqx_authz/src/emqx_authz_api_settings.erl
@@ -83,4 +83,6 @@ settings(put, #{
     {200, authorization_settings()}.

 authorization_settings() ->
-    maps:remove(<<"sources">>, emqx:get_raw_config([authorization], #{})).
+    C = maps:remove(<<"sources">>, emqx:get_raw_config([authorization], #{})),
+    Schema = #{roots => emqx_schema:fields("authorization"), fields => #{}},
+    hocon_tconf:make_serializable(Schema, C, #{}).
```

edit: maybe it's a good idea to define the authz settings as `emqx_schema:authz_settings()`,
having to use a string (`"authorization"`) to reference it is not quite friendly for static code analysis tools.

**Comment by zmstone** [comment]
need a rebalse.

**Comment by id** [comment]
@zmstone still not 100% comfortable with how schema and hocon work together, did I get your suggestion right?

I used `authz_fields` instead of `authz_settings` as you suggested, `_fields` seems to be more inline to other api functions there. Or do you think we should be `authz_settings` after all?

**Review by zmstone — APPROVED** [review]

**Pull request merged**